src/rules/alignRule.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING_SUFFIX = " are not aligned";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        var alignWalker = new AlignWalker(sourceFile, this.getOptions());
+        return this.applyWithWalker(alignWalker);
+    }
+}
+
+type SourcePosition = {
+    line: number;
+    character: number;
+}
+
+class AlignWalker extends Lint.RuleWalker {
+    public visitFunctionDeclaration(node: ts.FunctionDeclaration) {
+        this.checkParameters(node.parameters);
+        super.visitFunctionDeclaration(node);
+    }
+
+    public visitFunctionExpression(node: ts.FunctionExpression) {
+        this.checkParameters(node.parameters);
+        super.visitFunctionExpression(node);
+    }
+
+    public visitCallExpression(node: ts.CallExpression) {
+        this.checkArguments(node.arguments);
+        super.visitCallExpression(node);
+    }
+
+    public visitBlock(node: ts.Block) {
+        this.checkStatements(node.statements);
+        super.visitBlock(node);
+    }
+
+    private checkParameters(parameters: ts.Node[]) {
```

Comment:
I don't think you need these three private methods here -- you can probably just inline "this.checkAlignment("parameters", node.parameters)", etc, in all the above functions.